dll/win32/shell32/CShellLink.cpp

```diff
@@ -746,6 +746,25 @@ HRESULT STDMETHODCALLTYPE CShellLink::Load(IStream *stm)
     if (FAILED(hr)) // FIXME: Should we fail?
         return hr;
 
+    if (DATABLOCK_HEADER *pDB = SHFindDataBlock(m_pDBList, EXP_SPECIAL_FOLDER_SIG))
+    {
+        EXP_SPECIAL_FOLDER &special = *(EXP_SPECIAL_FOLDER*)pDB;
```

Comment:
Why not just
```suggestion
    pDB = (LPEXP_SPECIAL_FOLDER)SHFindDataBlock(m_pDBList, EXP_SPECIAL_FOLDER_SIG);
    if (pDB)
    {
```
directly, as done in the other parts of the file?
And then `pDB->xxx`, instead of `special.xxx` .